### DISCUSSION TITLE
Ansys Icepak "specified_version" Issue in n Icepak method of PYAEDT module

### DISCUSSION THREAD

**Discussion by Swapnillokam29** [ansys/pyaedt#4041] [Q&A] [answered]
I have installed Ansys Icepak 2021 R1 version 
![image](https://github.com/ansys/pyaedt/assets/156055091/5b30ec75-1854-4aed-8c18-fc8e998003fe)

I want to use PYAEDT Library to launch the Icepak Instance

I am using the following sample code to do the same 

import pyaedt
import os

temp_folder = pyaedt.generate_unique_folder_name()
package_temp_name = pyaedt.downloads.download_icepak(temp_folder)

ipk = pyaedt.Icepak(projectname=package_temp_name, **specified_version=211**, non_graphical=False, new_desktop_session=True)

ipk.autosave_disable

But I am getting the error as "Specified version 2021.1 is not installed on your system"

**-------------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------**
self._desktop_class = _init_desktop_from_design(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\desktop.py", line 379, in _init_desktop_from_design
    return Desktop(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\desktop.py", line 535, in __init__
    student_version_flag, version_key, version = self._assert_version(specified_version, student_version)
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\desktop.py", line 774, in _assert_version
    raise ValueError(
ValueError: Specified version 2021.1 is not installed on your system

**-------------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------**

As per the definition of Icepak() method in PYAEDT 

**specified_version** : str, int, float, optional
    Version of AEDT to use. The default is None, in which case the active version or latest installed version is used. This parameter is ignored when Script is launched within AEDT. Examples of input values are 232, 23.2,2023.2,"2023.2".
    
   So if I try to not include Specified Version parameter, It should open the default installed version of Icepak 
   
   But I am getting the type cast error that **"could not convert string to float: ''**
   
**-----------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------**
import pyaedt
import os

temp_folder = pyaedt.generate_unique_folder_name()
package_temp_name = pyaedt.downloads.download_icepak(temp_folder)

ipk = pyaedt.Icepak(projectname=package_temp_name ,non_graphical=False ,new_desktop_session=True)

ipk.autosave_disable

**-------------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------**
   
   Initializing new desktop!
PyAEDT INFO: StdOut is enabled
PyAEDT INFO: AEDT file load (threaded) time: 0.009456157684326172
PyAEDT INFO: Log on file is enabled
PyAEDT INFO: Log on Desktop Message Manager is enabled
PyAEDT INFO: Debug logger is disabled. PyAEDT methods will not be logged.
C:\Program Files\Python311\Lib\site-packages\pyaedt\misc\misc.py:14: UserWarning: No installed versions of AEDT are found in the system environment variables ``ANSYSEM_ROOTxxx``.
  warnings.warn(
Traceback (most recent call last):
  File "d:\Swapnil_Learning\Cool_Sim\IcepakEntireRepo\pyaedt\examples\04-Icepak\Sample.py", line 7, in <module>
    ipk = pyaedt.Icepak(projectname=package_temp_name ,non_graphical=False ,new_desktop_session=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\generic\design_types.py", line 385, in Icepak
    return app(
           ^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\icepak.py", line 152, in __init__
    FieldAnalysis3D.__init__(
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\application\Analysis3D.py", line 77, in __init__
    Analysis.__init__(
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\application\Analysis.py", line 111, in __init__
    Design.__init__(
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\application\Design.py", line 227, in __init__
    self._desktop_class = _init_desktop_from_design(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\desktop.py", line 379, in _init_desktop_from_design
    return Desktop(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\desktop.py", line 535, in __init__
    student_version_flag, version_key, version = self._assert_version(specified_version, student_version)
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\site-packages\pyaedt\desktop.py", line 766, in _assert_version
    if float(specified_version[0:6]) < 2019:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: **"could not convert string to float: ''**
   **-----------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------**

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @Swapnillokam29 ,

I think you are trying to control Icepak Classic, which is not the same as AEDT Icepak. PyAEDT only controls AEDT solutions (HFSS, Maxwell, Icepak AEDT, Mechanical AEDT...)

Apart from that, PyAEDT is compatible with AEDT 2022R2, and I suggest updating to the latest version 2023R2:
https://aedt.docs.pyansys.com/version/stable/Getting_started/About.html#dependencies

Regards,

**Reply by Swapnillokam29** [reply]
@Samuelopez-ansys 
Thanks for the clarity and your time
